src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -95,6 +97,102 @@ CHIP_ERROR GeneralCommissioningAttrAccess::Read(const ConcreteReadAttributePath
     case SupportsConcurrentConnection::Id: {
         return ReadSupportsConcurrentConnection(aEncoder);
     }
+#if CHIP_CONFIG_TC_REQUIRED
+    case TCAcceptedVersion::Id: {
+        TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+        Optional<TermsAndConditions> outTermsAndConditions;
+        CHIP_ERROR err;
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        err = termsAndConditionsProvider->GetAcceptance(outTermsAndConditions);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        return !outTermsAndConditions.HasValue() ? aEncoder.Encode(static_cast<uint16_t>(0))
+                                                 : aEncoder.Encode(outTermsAndConditions.Value().version);
+    }
+    case TCMinRequiredVersion::Id: {
+        TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+        Optional<TermsAndConditions> outTermsAndConditions;
+        CHIP_ERROR err;
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        err = termsAndConditionsProvider->GetRequirements(outTermsAndConditions);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
```

Comment:
For these returns on conditions not met or err != CHIP_NO_ERROR, we use `VerifyOrReturnError(cond, error)` and `ReturnErrorOnFailure(err)`

It would make the overall vertical space in here much tighter.

Reply:
I tried to update all of the error checks to use either `VerifyOrReturnError`, `VerifyOrReturnValue`, or `ReturnErrorOnFailure`. Let me know if this is how you wanted it.